coordinator/src/storage.rs

```diff
@@ -0,0 +1,108 @@
+use bitcoin::BlockHash;
+use lightning::chain::channelmonitor::ChannelMonitor;
+use lightning::sign::EntropySource;
+use lightning::sign::SignerProvider;
+use lightning::util::persist::KVStorePersister;
+use lightning::util::ser::Writeable;
+use lightning_persister::FilesystemPersister;
+use ln_dlc_node::storage::LDKStoreReader;
+use ln_dlc_node::storage::TenTenOneStorage;
+use ln_dlc_storage::sled::SledStorageProvider;
+use ln_dlc_storage::DLCStoreProvider;
+use std::fs;
+use std::ops::Deref;
+use std::path::Path;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct TenTenOneNodeStorage {
+    pub ln_storage: Arc<FilesystemPersister>,
+    pub dlc_storage: Arc<SledStorageProvider>,
+    pub data_dir: String,
+}
+
+impl TenTenOneStorage for TenTenOneNodeStorage {}
+
+impl TenTenOneNodeStorage {
+    pub fn new(data_dir: String) -> TenTenOneNodeStorage {
+        let data_dir = PathBuf::from(data_dir);
+
+        if !data_dir.exists() {
+            fs::create_dir_all(data_dir.as_path())
+                .unwrap_or_else(|_| panic!("Failed to create data dir {data_dir:?}"));
```

Comment:
What's the reason for panicing here and not returning an error? (same for rest of the file).

Reply:
hmm.. I guess if we can't create the data directory, there is not much we can do - so a panic seems reasonable to me.